common/ts/prepare-task.ts

```diff
@@ -44,6 +44,9 @@ export default async function prepareTask(endpoint: Endpoint, rootPath: string)
     .map(keyValue => keyValue.split(/=(.+)/))
     .forEach(([k, v]) => (props[k] = v));
 
+  // Save BuildId so that the scan can later uniquely be identified in "project_analyses API" or similar
+  props['sonar.buildid'] = tl.getVariable('Build.BuildId');
```

Comment:
Note that I have no idea what key the buildid should be saved to. In this PR I just assume the key would be called `sonar.buildid`.